yb-voyager/cmd/importData.go

```diff
@@ -886,13 +887,80 @@ func doOneImport(task *SplitFileImportTask, connPool *tgtdb.ConnectionPool) {
 	markTaskDone(task)
 }
 
-func importSplit(conn *pgx.Conn, task *SplitFileImportTask, fh *os.File, copyCmd string) (int64, error) {
+func importSplit(conn *pgx.Conn, task *SplitFileImportTask, fh *os.File, copyCmd string) (rowsAffected int64, err error) {
 	// reset the reader to begin for every call
 	fh.Seek(0, io.SeekStart)
 	//setting the schema so that COPY command can acesss the table
 	setTargetSchema(conn)
-	res, err := conn.PgConn().CopyFrom(context.Background(), fh, copyCmd)
-	return res.RowsAffected(), err
+
+	// NOTE: DO NOT DEFINE A NEW err VARIABLE IN THIS FUNCTION. ELSE, IT WILL MASK THE err FROM RETURN LIST.
+	ctx := context.Background()
+	var tx pgx.Tx
+	tx, err = conn.BeginTx(ctx, pgx.TxOptions{})
+	if err != nil {
+		return 0, fmt.Errorf("begin transaction: %w", err)
+	}
+	defer func() {
+		var err2 error
+		if err != nil {
+			err2 = tx.Rollback(ctx)
+			if err2 != nil {
+				rowsAffected = 0
+				err = fmt.Errorf("rollback txn: %w (while processing %s)", err2, err)
+			}
+		} else {
+			err2 = tx.Commit(ctx)
+			if err2 != nil {
+				rowsAffected = 0
+				err = fmt.Errorf("commit txn: %w", err2)
+			}
+		}
+	}()
+
+	// Check if the split is already imported.
+	var alreadyImported bool
+	alreadyImported, rowsAffected, err = splitIsAlreadyImported(task, tx)
+	if err != nil {
+		return 0, err
+	}
+	if alreadyImported {
+		return rowsAffected, nil
+	}
+
+	// Import the split using COPY command.
+	var res pgconn.CommandTag
+	res, err = tx.Conn().PgConn().CopyFrom(context.Background(), fh, copyCmd)
```

Comment:
Is this COPY indeed part of the transaction?

Reply:
@amit-yb did you get a chance to verify this?